unblob/processing.py

```diff
@@ -289,6 +288,114 @@ def _process_task(self, result: TaskResult, task: Task):
         _FileTask(self._config, task, stat_report.size, result).process()
 
 
+class _DirectoryTask:
+    def __init__(self, config: ExtractionConfig, dir_task: Task, result: TaskResult):
+        self.config = config
+        self.dir_task = dir_task
+        self.result = result
+
+    def process(self):
+        if self.dir_task.multi_file:
+            # The file has been processed as part of a previous MultiFile
+            return
+
+        logger.debug("Processing directory", path=self.dir_task.path)
+
+        processed_paths: Set[Path] = set()
+        extract_dirs: Set[Path] = set()
+        for dir_handler_class in self.config.dir_handlers:
+            dir_handler = dir_handler_class()
+
+            for path in dir_handler.PATTERN.get_files(self.dir_task.path):
```

Comment:
This function is too big (~100 lines, multi-page for me) and complex. This for loop looks something that could be extracted to cut the function size in half.